staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -375,19 +375,18 @@ func (d *DiscoveryClient) SwaggerSchema(version schema.GroupVersion) (*swagger.A
 	return &schema, nil
 }
 
-// OpenAPISchema fetches the open api schema using a rest client and parses the json.
-// Warning: this is very expensive (~1.2s)
-func (d *DiscoveryClient) OpenAPISchema() (*spec.Swagger, error) {
-	data, err := d.restClient.Get().AbsPath("/swagger.json").Do().Raw()
+// OpenAPISchema fetches the open api schema using a rest client and parses the proto.
+func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
+	data, err := d.restClient.Get().AbsPath("/swagger-2.0.0.pb-v1").Do().Raw()
```

Comment:
> The plan is to change validation to use this code, but even then, one will be able to go back to the old swagger validation with a flag

It still seems capricious and unnecessary to break.  We are just talking about keeping the old code in the `if err != nil` block, right?



> If --validate=true, we should fail if we can't get the openapi to validate.

The only reason you "can't" is for want of a few lines right here.